czkawka_core/src/duplicate.rs

```diff
@@ -905,9 +909,10 @@ impl DuplicateFinder {
             self.information.bytes_read_when_hashing += bytes_read;
             self.text_messages.warnings.append(&mut errors);
             for (_hash, vec_file_entry) in hash_map {
-                if vec_file_entry.len() > 1 {
+                let dups = self.filter_hard_links(vec_file_entry);
+                if dups.len() > 1 {
                     self.files_with_identical_hashes.entry(size).or_insert_with(Vec::new);
-                    self.files_with_identical_hashes.get_mut(&size).unwrap().push(vec_file_entry);
+                    self.files_with_identical_hashes.get_mut(&size).unwrap().push(dups);
```

Comment:
This should be moved into `check_file_size` function, because for now this only works for full file hashing and checking files before hashing will allow to drop some records before doing such resource heavy operation.